Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -522,119 +524,31 @@ void MemoryWidget::ValidateAndPreviewInputValue()
 {
   m_data_preview->clear();
   QString input_text = m_data_edit->text();
-  const auto combo_id = static_cast<InputID>(m_input_combo->currentData().toInt());
+  const auto input_type = static_cast<Type>(m_input_combo->currentData().toInt());
 
-  m_base_check->setEnabled(combo_id == InputID::U32 || combo_id == InputID::S32 ||
-                           combo_id == InputID::U16 || combo_id == InputID::S16 ||
-                           combo_id == InputID::U8 || combo_id == InputID::S8);
+  m_base_check->setEnabled(input_type == Type::Unsigned32 || input_type == Type::Signed32 ||
+                           input_type == Type::Unsigned16 || input_type == Type::Signed16 ||
+                           input_type == Type::Unsigned8 || input_type == Type::Signed8);
 
   if (input_text.isEmpty())
     return;
 
   // Remove any spaces
-  if (combo_id != InputID::ASCII)
+  if (input_type != Type::ASCII)
     input_text.remove(QLatin1Char(' '));
 
   QFont font;
   QPalette palette;
   QString hex_string;
-  bool good = false;
-  const int radix = (m_base_check->isChecked() && m_base_check->isEnabled()) ? 16 : 0;
-
-  switch (combo_id)
-  {
-  case InputID::ASCII:
-  {
-    good = true;
-    const QByteArray bytes = input_text.toLatin1();
-    hex_string = QString::fromLatin1(bytes.toHex());
-    break;
-  }
-  case InputID::FLOAT:
-  {
-    const float value_float = input_text.toFloat(&good);
+  std::vector<u8> bytes = m_memory_view->ConvertTextToBytes(input_type, input_text);
 
-    if (good)
-    {
-      const u32 hex_out = Common::BitCast<u32>(value_float);
-      hex_string = QString::fromStdString(fmt::format("{:08X}", hex_out));
-    }
-    break;
-  }
-  case InputID::DOUBLE:
+  if (!bytes.empty())
   {
-    const double value_double = input_text.toDouble(&good);
+    std::string s;
+    for (const u8 c : bytes)
+      s.append(fmt::format("{:x}", c));
```

Comment:
Ah, here's the bug. This needs to be `{:02x}` so you always get two digits back.

Reply:
Ah thanks!